### DISCUSSION TITLE
Adding Functions to an ESP32 with ESPAltherma

### DISCUSSION THREAD

**Discussion by Juhe0167** [raomin/ESPAltherma#263] [Q&A]
Hello,
first of all a big thank you to all who are working on this project.
I got the ESPAltherma up and running very easily and it gives me good information about my heat pump.
I have an ETSXB16P50EF and an EPRA14DAW1 running.

Now I would like to add two new functions to the ESP32:
Up to 8 thermostats with DS18B20 and Powermeter with Pulse Counter.
I found some libraries in Platformio and some how-tos on the web, but the how-tos are made for a single use of the functions.

Now I would ask for help to get things together because I am totally new with ESP and can't filter things to do it right.

Maybe the result will be interesting for others then, as I think the ESP32 is performant enough to do these tasks.

Best regards
Jürgen

**Comment by raomin** [comment]
Where do you want to add 8 thermostats? On your heatpump? If so, then yes, your esp32 with ESPAltherma can be extended to manage them on an extra pin (you can daisy chain multiple DS18B20 on one data wire).
But if you are _totally new_ I would suggest to add a new esp32 with _[ESPHome](https://esphome.io/components/sensor/dallas.html)_ (or multiple ones if your 8 thermostats would be in different rooms)

**Comment by Juhe0167** [comment]
Hello raomin,
the temperatures that should be taken are all in one room and in the heatpumproom.
Thats why I'm thinking about integrating it into the ESPAlteherma ESP32.

The pulse from the powermeter ist some meters away, but I have already pulled a Cable for it.

The only thing I do not have by now is a 4k7-resistor...

**Comment by Juhe0167** [comment]
Hello,
can someone give me some startingpoints to integrate extrenal Temperature an S0-Pulse in ESPAltherma?
I want to try to do so, but I need some help for the beginning.
Thanks.
